src/main/java/listeners/actions/SampleBlockActionListener.java

```diff
@@ -7,47 +7,58 @@
 import static com.slack.api.model.view.Views.*;
 
 import com.google.gson.Gson;
-import com.slack.api.app_backend.interactive_components.payload.BlockActionPayload;
+import com.slack.api.bolt.App;
 import com.slack.api.bolt.context.builtin.ActionContext;
 import com.slack.api.bolt.handler.builtin.BlockActionHandler;
 import com.slack.api.bolt.request.builtin.BlockActionRequest;
 import com.slack.api.bolt.response.Response;
 import com.slack.api.methods.SlackApiException;
-import com.slack.api.methods.response.views.ViewsUpdateResponse;
 import com.slack.api.model.view.View;
 import com.slack.api.util.json.GsonFactory;
 import java.io.IOException;
-import java.util.Map;
+import models.Constants;
+import models.MeetingArrangement;
 
 public class SampleBlockActionListener implements BlockActionHandler {
+    private final App app;
+    private final Gson gson;
+
+    public SampleBlockActionListener(App app) {
+        this.app = app;
+        this.gson = GsonFactory.createSnakeCase(app.getSlack().getConfig());
+    }
 
     @Override
     public Response apply(BlockActionRequest req, ActionContext ctx) throws IOException, SlackApiException {
-        BlockActionPayload payload = req.getPayload();
-
-        String categoryId = payload.getActions().get(0).getSelectedOption().getValue();
-        View currentView = payload.getView();
-
-        String privateMetadata = currentView.getPrivateMetadata();
-        View viewForTheCategory = buildViewByCategory(categoryId, privateMetadata);
-
-        ViewsUpdateResponse viewsUpdateResp = ctx.client().viewsUpdate(r -> r.viewId(currentView.getId())
-                .hash(currentView.getHash())
-                .view(viewForTheCategory));
+        this.app.executorService().submit(() -> {
```

Comment:
Using executor service is a common practice to deal with 3 second timeouts and it is a very common question from communities. Having ready-to-go examples should be useful for many devs.

Reply:
I was not aware of this solution, thank you for bringing it up 🥇